react/features/base/aspect-ratio/components/AspectRatioDetector.native.js

```diff
@@ -1,73 +1,85 @@
-import PropTypes from 'prop-types';
-import React, { Component } from 'react';
-import { View } from 'react-native';
+// @flow
+
+import { Component } from 'react';
+import { DeviceEventEmitter, Dimensions } from 'react-native';
 import { connect } from 'react-redux';
 
 import { setAspectRatio } from '../actions';
-import styles from './styles';
+
+/**
+ * AspectRatioDetector component's property types.
+ */
+type Props = {
+
+    /**
+     * Any nested components.
+     */
+    children: React$Node,
+
+    /**
+     * Redux store dispatch function.
+     */
+    dispatch: Dispatch<*>
+}
 
 /**
  * A root {@link View} which captures the 'onLayout' event and figures out
  * the aspect ratio of the app.
  */
-class AspectRatioDetector extends Component {
+class AspectRatioDetector extends Component<Props> {
+    /**
+     * Listener for orientation changes.
+     */
+    _orientationListener: Object;
+
+    /**
+     * Implements React's {@link Component#componentWillMount()}. Invoked
+     * immediately before mounting occurs. Add a listener for orientation
+     * changes.
+     *
+     * @inheritdoc
+     */
+    componentWillMount() {
+        this._orientationListener = DeviceEventEmitter.addListener(
+            'namedOrientationDidChange', () => {
+                this._setAspectRatio();
+            }
+        );
+
+        // Get the initial orientation
+        this._setAspectRatio();
```

Comment:
well I don't know... the feature was about the width and height dimensions, but this even will be triggered only when the orientation will change

Reply:
I'm happier detecting the orientation with a built-in library that actually detects it based on hardware, other than based on aspect ratio. But then I'm not sure why we need the extra component. Can't we use something that actually uses hardware info and not tied to an additional component?

Reply:
I see that in the other PR you need that, because the onLayout does not trigger, how about using both at the same time ? Otherwise the whole "aspect ratio" feature would have to be changed to "orientation" and all naming related to that feature

Reply:
The point is what really matters is width vs height and not just the orientation, because that's the space you have for drawing things. There are situations when screen can be narrow when in landscape mode.

Reply:
Aaaaah, wait! That last thing you said is absolutely true @paweldomas !!! When using multi-tasking on the iPad this can happen, IIRC. We may need to rethink this :-(

Reply:
Ok, then I put that attribute back to the settings screen till then, if you don't mind, so then you have time to rethink this one. Then we can remove it when it's solved

Reply:
Please leave a comment so we don't forget.